sklearn/metrics/cluster/_supervised.py

```diff
@@ -138,6 +146,206 @@ def contingency_matrix(labels_true, labels_pred, *, eps=None, sparse=False):
 
 # clustering measures
 
+def pair_confusion_matrix(labels_true, labels_pred):
+    """Pair confusion matrix arising from two clusterings.
+
+    The pair confusion matrix computes a 2x2 similarity matrix between
+    two clusterings by considering all pairs of samples and counting
+    pairs that are assigned in the same or different clusters in the
+    predicted and true clusterings.
+
+    The 2x2 pair confusion matrix::
+
+        D00 D01
+        D10 D11
+
+    has the following entries
+
+    D00
+      number of pairs with both clusterings having the samples
+      not clustered together
+
+    D10
+      number of pairs with the true label clusterings having the
+      samples clustered together but the other clustering not
+      having the samples clustered together
+
+    D01
+      number of pairs with the true label clusterings not having
+      the samples clustered together but the other clustering
+      having the samples clustered together
+
+    D11
+      number of pairs with both clusterings having the samples
+      clustered together
+
+    Parameters
+    ----------
+    labels_true : int array, shape = [n_samples]
+        Ground truth class labels to be used as a reference
+
+    labels_pred : array, shape = [n_samples]
+        Cluster labels to evaluate
+
+    Returns
+    -------
+    nested array [[D00, D01], [D10, D11]]
+
+    Examples
+    --------
+    Perfectly matching labelings have all non-zero entries on the
```

Comment:
Please give just one or two examples here, not detailed instruction.

Reply:
Done.